src/project-type/workspace-instance.ts

```diff
@@ -0,0 +1,34 @@
+import { WorkspaceApi, WorkspaceImpl } from "@sap/project-api";
+import { BasWorkspaceApi } from "../../types/api";
+
+let basWSAPI: BasWorkspaceApi;
```

Comment:
- I think we need to keep a ref to the `WorkspaceImpl` for our internal logic
- I am not sure we need to keep a ref to the "read-only" version.